pep-0561.rst

```diff
@@ -42,40 +43,73 @@ checkers to fail on packages that are highly dynamic (e.g. sqlalchemy
+
+"inline" - the types are part of the runtime code using PEP 526 and 3107 
+syntax.
+
+"stubs" - files containing only type information, empty of runtime code.
+
+"Distributions" are the packaged files which are used to publish and distribute
+a release. [3]_
+
+"Module" a file containing Python runtime code or stubbed type information.
+
+"Package" a directory or directories that namespace Python modules.
+
+
 Specification
 =============
 
 There are several motivations and methods of supporting typing in a package.
-This PEP recognizes three (3) types of packages that may be created:
+This PEP recognizes three (3) types of packages that users of typing wish to
+create:
 
 1. The package maintainer would like to add type information inline.
 
 2. The package maintainer would like to add type information via stubs.
 
-3. A third party would like to share stub files for a package, but the
-   maintainer does not want to include them in the source of the package.
+3. A third party or package maintainer would like to share stub files for
+   a package, but the maintainer does not want to include them in the source
+   of the package.
    
 This PEP aims to support these scenarios and make them simple to add to
 packaging and deployment.
 
 The two major parts of this specification are the packaging specifications
-and the resolution order for resolving module type information. The packaging
-spec is based on and extends PEP 345 metadata. The type checking spec is
-meant to replace the ``shared/typehints/pythonX.Y/`` spec of PEP 484 [2]_.
+and the resolution order for resolving module type information. The type
+checking spec is meant to replace the ``shared/typehints/pythonX.Y/`` spec
+of PEP 484 [2]_.
+
+New third party stub libraries SHOULD distribute stubs via the third party
+packaging methods proposed in this PEP in place of being added to typeshed.
+Typeshed will remain in use, but if maintainers are found, third party stubs
+in typeshed MAY be split into their own package.
 
-New third party stub libraries are encouraged to distribute stubs via the
-third party packaging proposed in this PEP in place of being added to 
-typeshed. Typeshed will remain in use, but if maintainers are found, third
-party stubs in typeshed are encouraged to be split into their own package.
 
 Packaging Type Information
 --------------------------
+
 In order to make packaging and distributing type information as simple and
-easy as possible, the distribution of type information, and typed Python code
-is done through existing packaging frameworks. This PEP adds a new item to the
-``*.distinfo/METADATA`` file to contain metadata about a package's support for
-typing. The new item is optional, but must have a name of ``Typed`` and have a
-value of either ``inline`` or ``stubs``, if present.
+easy as possible, packaging and distribution is done through existing
+frameworks.
+
+To enable the safe distribution of typing information, packaging tools MUST
```

Comment:
If I've understood your revised proposal correctly, this isn't quite the right framing, as packaging tools don't need to do anything new, packaging *publishers* are being asked to include a marker file inside their package directories that can be accessed with https://docs.python.org/3/library/pkgutil.html#pkgutil.get_data given the location of the package itself.

For a package published with `setuptools`, that will look like the following [package_data](https://packaging.python.org/tutorials/distributing-packages/#package-data) entry:

    package_data={
        'pkgname': ['py.typeinfo'],
    },

(I know I originally advised *against* mentioning `setuptools` APIs in the PEP, but with the revised proposal, I think it will make it clearer that you're proposing to use existing packaging system features, rather than proposing new ones)

Reply:
I was originally thinking the maintainer would indicate through the packaging tool (such as a new keyword to `setup()`) that the package supported typing (through a dict mapping package names to the type of information). Then it would be the responsibility of the tooling to install a generated `py.typeinfo` for listed packages.

However, your proposal has quite a few advantages. It requires no change to tooling, and should work on all Python versions back to 2.7, if I am correct. I am slightly leery of putting the onus of making the `py.typeinfo` on the maintainer however, as in the current spec the format is in the METADATA style, which is meant for tooling to both read and write AIUI. Perhaps the contents of the file can be changed to be either plain `inline` or `stubs`. I'm interested in your thoughts on this.

Reply:
Tools may be updated to make this easier (such as with a `type_hints=True` parameter), but it's better for the PEP itself to not *rely* on tooling updates (since it complicates the implementation).

As far as the format of the file itself goes, you may even want to encode the setting in the filename itself (`py.inline.typeinfo` vs `py.stubs.typeinfo`), such that the setting can be checked with os.listdir(), rather than having to actually open and read multiple files in multiple directories (which can be very slow, especially on network drives).

Reply:
Yes, I was considering putting it in the file name as well. I will update the text now. Thank you for the feedback, it was very helpful!